SonarQube.Common/CommandLine/CommandLineParser.cs

```diff
@@ -0,0 +1,182 @@
+﻿//-----------------------------------------------------------------------
+// <copyright file="CommandLineParser.cs" company="SonarSource SA and Microsoft Corporation">
+//   Copyright (c) SonarSource SA and Microsoft Corporation.  All rights reserved.
+//   Licensed under the MIT License. See License.txt in the project root for license information.
+// </copyright>
+//-----------------------------------------------------------------------
+
```

Comment:
Extracted to a separate file and moved to Common. This class parses a set of command line according to a supplied list of descriptors that specify what the acceptable parameters are. It logs errors for missing, duplicate and unknown arguments.